src/lib.rs

```diff
@@ -36,13 +38,10 @@ pub struct TodoOperation {
     todo: TodoData,
 }
 
-pub trait ToOperation {
-    fn to_operation_string(&self) -> String;
-}
-
-impl ToOperation for TodoOperation {
-    fn to_operation_string(&self) -> String {
-        format!(
+impl Display for TodoOperation {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(
```

Comment:
Just curious, why `write!` vs `format!`?

Reply:
`format!` allocates and returns a string, whereas `write!` simply says "given a format string, generate the correct stringified-output-into a target buffer, whatever that buffer is" ([docs](http://doc.rust-lang.org/1.56.0/std/macro.write.html)). The buffer could be a stdio pipe, or a string, or a stream, or a socket, or whatever.